server/src/main/java/io/druid/segment/realtime/appenderator/AppenderatorImpl.java

```diff
@@ -1026,6 +1093,7 @@ public String apply(SegmentIdentifier input)
     // Decrement this sink's rows from rowsCurrentlyInMemory (we only count active sinks).
     rowsCurrentlyInMemory.addAndGet(-sink.getNumRowsInMemory());
     totalRows.addAndGet(-sink.getNumRows());
+    bytesCurrentlyInMemory.addAndGet(-sink.getBytesInMemory());
```

Comment:
This and `rowsCurrentlyInMemory.addAndGet(-sink.getNumRowsInMemory());` are not necessary because sink should be pushed before calling this method and thus there should be no data in memory. Instead, we need to add a sanity check that `sink.getNumRowsInMemory() == 0`. Same for `getBytesInMemory()`.

Reply:
Not clear, where do we want to add this sanity check ?